lib/elixir/lib/protocol.ex

```diff
@@ -226,6 +226,14 @@ defmodule Protocol do
   end
 
   @doc """
+  Returns a list of implementations, or `:error` if the protocol was not consolidated.
+  """
+  @spec impls(module) :: [module] | :error
```

Comment:
100% sure you're right about this, but would you mind quickly explaining why we're going with the callback one over this one? Just curious. :)